src/store/kv-block-store.ts

```diff
@@ -0,0 +1,205 @@
+/**
+ * AR.IO Gateway
+ * Copyright (C) 2022-2023 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import winston from 'winston';
+
+import {
+  fromB64Url,
+  jsonBlockToMsgpack,
+  msgpackToJsonBlock,
+  toB64Url,
+} from '../lib/encoding.js';
+import { sanityCheckBlock } from '../lib/validation.js';
+import { KVBufferStore, PartialJsonBlockStore } from '../types.js';
+import { PartialJsonBlock } from '../types.js';
+
+export class KvBlockStore implements PartialJsonBlockStore {
+  private log: winston.Logger;
+  private kvBufferStore: KVBufferStore;
+
+  constructor({
+    log,
+    kvBufferStore,
+  }: {
+    log: winston.Logger;
+    kvBufferStore: KVBufferStore;
+  }) {
+    this.log = log.child({ class: this.constructor.name });
+    this.kvBufferStore = kvBufferStore;
+  }
+
+  private blockHash(hash: string) {
```

Comment:
Mostly just a style preference, but I'd probably pull these out of the class itself (but don't export them) and rename them to `heightKey` and `hashKey`.

Reply:
[66f4cf7](https://github.com/ar-io/ar-io-node/pull/60/commits/66f4cf7800807ed7f587e875fbde7653a103bb18)